transport/src/main/java/io/camunda/zeebe/transport/stream/api/ClientStreamConsumer.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.transport.stream.api;
+
+import io.camunda.zeebe.util.buffer.BufferReader;
+
+/**
+ * Represents a consumer of a stream which can consume data of type P. The data is typically pushed
+ * from a broker, and consumed by gateway.
+ *
+ * @param <P> the payload type that can be pushed to the stream
+ */
+public interface ClientStreamConsumer<P extends BufferReader> {
```

Comment:
This is currently just a placeholder. It will be changed when we add actual handling of job received from the stream.

Reply:
@npepinpe This interface is also changed from our prototype. I remove streamType and metadata out of it.